src/hypertable.c

```diff
@@ -1918,7 +1922,33 @@ ts_hypertable_create_internal(PG_FUNCTION_ARGS, bool is_dist_call)
 	ht = ts_hypertable_cache_get_cache_and_entry(table_relid, CACHE_FLAG_NONE, &hcache);
 
 	if (NULL != space_dim_info)
+	{
+		/* Function `ts_hypertable_create_from_info` will error if the
+		 * hypertable already exists unless the if-not-exists flag is not set.
+		 * If we reach this point, either if-not-exists-flag was set or the
+		 * hypertable did not exist and was created above.
+		 *
+		 * In `ts_hypertable_create_from_info`, a call to
+		 * `ts_dimension_info_validate(space_dim_info)` will be made if (and
+		 * only if) the table did not exist. The function does not only
+		 * validate the dimension, it also set `dimension_id` field.
+		 *
+		 * If the table already existed, `created` will be false and the
+		 * `dimension_id` will be set to the invalid OID, which means that
+		 * `ts_hypertable_check_partitioning` will crash since it it expect a
+		 * proper OID to be passed. This means that we have to fetch the
+		 * dimension OID here in the event that the table was not created. */
+		if (!created)
+		{
+			Dimension *dim = ts_hyperspace_get_dimension_by_name(ht->space,
+																 DIMENSION_TYPE_CLOSED,
+																 NameStr(*space_dim_name));
+			Assert(dim);
+			space_dim_info->dimension_id = dim->fd.id;
```

Comment:
This doesn't make much sense. Why do we create `space_dim_info` then fill in dimension_id here if the table already existed? Why even validate it and check partitioning in that case?